src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,357 @@
+    dc_xlower = value[0] * xlower[0] + value[1] * ylower[0]
+    dc_yupper = value[0] * xupper[1] + value[1] * yupper[1]
+    dc_ylower = value[0] * xlower[1] + value[1] * ylower[1]
+
+    currUX = DCREVERSCOEF * DCROTUPPXX * dc_xupper
+    currLX = DCREVERSCOEF * DCROTLOWXX * dc_xlower
+    currUY = DCREVERSCOEF * DCROTUPPYY * dc_yupper
+    currLY = DCREVERSCOEF * DCROTLOWYY * dc_ylower
+
+    for current in [currLX, currLY, currUX, currUY]:
+        if current < C_MIN_DBL_SHIFT or current > C_MAX_DBL_SHIFT:
+            raise ValueError("Beam deflection %s exceeds limits (%s, %s) of DC coils."
+                             % (current, C_MIN_DBL_SHIFT, C_MAX_DBL_SHIFT))
+
+    rawLX = current_to_raw(currLX)
+    rawUX = current_to_raw(currUX)
+    rawLY = current_to_raw(currLY)
+    rawUY = current_to_raw(currUY)
+
+    return [rawLX, rawLY, rawUX, rawUY]
+
+
+def transform_coordinates_reverse(register_values, xlower, ylower, xupper, yupper):
+    """
+    Transform register values back to x, y position in source coordindate system.
+    :param register_values: (int, int, int, int) register values
+    :param xlower: (float, float) xlower metadata
+    :param xupper: (float, float) xupper metadata
+    :param ylower: (float, float) ylower metadata
+    :param yupper: (float, float) yupper metadata
+    :return (int, int): x, y position
+    """
+    rawLX, rawLY, rawUX, rawUY = register_values
+
+    currLX = raw_to_current(rawLX)
+    currLY = raw_to_current(rawLY)
+    currUX = raw_to_current(rawUX)
+    currUY = raw_to_current(rawUY)
+
+    dc_xupper = currUX / (DCREVERSCOEF * DCROTUPPXX)
+    dc_xlower = currLX / (DCREVERSCOEF * DCROTLOWXX)
+    dc_yupper = currUY / (DCREVERSCOEF * DCROTUPPYY)
+    dc_ylower = currLY / (DCREVERSCOEF * DCROTLOWYY)
+
+    # Now we have to solve an overdetermined linear system of four equations with two variables.
+    A = numpy.array([[xupper[0], yupper[0]], [xlower[0], ylower[0]],
+                     [xupper[1], yupper[1]], [xlower[1], ylower[1]]])
+    b = numpy.array([dc_xupper, dc_xlower, dc_yupper, dc_ylower])
+    value, *_ = numpy.linalg.lstsq(A, b)
+
+    # A = numpy.array([[xupper[0], yupper[0]], [xlower[0], ylower[0]]])
+    # b = numpy.array([dc_xupper, dc_xlower])
+    # value = numpy.linalg.solve(A, b)
+    value = (value[0] * 1e-6, value[1] * 1e-6)  # µm --> m
+    return value
+
+
+class BeamShiftController(model.HwComponent):
+    """
+    Driver for the Thermofischer beam deflection controller.
+    This class provides the .shift VA containing a tuple of two floats which describe
+    the x and y beam offset in m in the stage coordinate system.
+
+    The conversion to internal ampere values (including scaling and rotation) is specified
+    through the MD_CALIB_BEAMSHIFT metadata (a 4x2 tuple, 4x (float, float)).
+    """
+
+    def __init__(self, name, role, port=None, serialnum=None, **kwargs):
+        """
+        :param port (str): port (e.g. /dev/ttyUSB0)
```

Comment:
Please indicate here the docstring from findDevice (about the special value `/dev/fake` and pattern).